### PR TITLE
162633504-ft(Posting Comment): Implements Posting a comment.

### PR DESCRIPTION
**Pull request by araaliFarooq**
## What does this PR do?
* Implements functionality that enables users to comment on different articles .

### Description of Tasks completed
* Enables users to comment on different articles .
* Fixes email conflict errors in users social authentication
* Enables users to edit and delete their comments
* Enables a user to view just one specific comment's details.

### How to manually test the above functionality.
* Run migrations to create the Comment model
* #### `python manage.py test` to run tests
#### Endpoints to test the feature
HTTP Method|End point | Public Access|Action | Request Body
-----------|----------|--------------|---------------------|--------------- 
POST | /api/v1/articles/test-article-today/comments/ | False | Post a comment | fig-1
GET | /api/v1/articles/test-article-today/comments/  | FALSE | View all comments for an article
GET | /api/v1/articles/test-article-today/comments/<int:pk> | False | View single comment
PUT | /api/v1/articles/test-article-today/comments/<int:pk>| False | Edit single comment | fig-2
DELETE | /api/v1/articles/test-article-today/comments/<int:pk>| False | Delete single comment

#### fig-1
* {
  "comment": {
    "body": " This is a sample comment."
  }
}

#### fig-2
*  {
    "body": " This is a sample comment update."
  }

#### [Delivers #162633504]
* https://www.pivotaltracker.com/story/show/162633504

### CODE REVIEW

File: authors/apps/comments/api_views.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    lookup_field = "slug"
+    permission_classes = (permissions.IsAuthenticatedOrReadOnly,)
+
+    def post(self, request, *args, **kwargs):
+        comment = request.data.get("comment", {})
+        serializer = CommentList.serializer_class(data=comment)
+        serializer.is_valid(raise_exception=True)
+        user = request.user
+        slug = kwargs["slug"]
+        articles = Article.objects.filter(slug=slug).first()
```
Comment (jeanjoe) [inline authors/apps/comments/api_views.py:28]: Please add a check when the article doesn't exist

**Review by jeanjoe — DISMISSED** [review]

File: authors/apps/comments/api_views.py (hunk 1 above)
Comment (araaliFarooq) [reply authors/apps/comments/api_views.py:28]: Thanks @jeanjoe , I have implemented that check.

**Comment by fatukunda** [comment]
LGTM

**Comment by mubstimor** [comment]
LGTM

**Comment by Qadriyah** [comment]
Good work done however, you need to display the comment with the article. You should add a comment field on the article model.

**Comment by araaliFarooq** [comment]
> Good work done however, you need to display the comment with the article. You should add a comment field on the article model.

Thanks Baker for your feedback. I have resolved it now.

**Comment by jeanjoe** [comment]
LGTM

**Comment by Qadriyah** [comment]
LGTM

**Pull request merged**